File: spec/java/src/io/kaitai/struct/spec/TestParamsCallShort.java
package io.kaitai.struct.spec;

import io.kaitai.struct.testformats.ParamsCallShort;
import org.testng.annotations.Test;

import static org.testng.Assert.assertEquals;

public class TestParamsCallShort extends CommonSpec {
    @Test
    public void testParamsCallShort() throws Exception {
        ParamsCallShort r = ParamsCallShort.fromFile(SRC_DIR + "term_strz.bin");

        assertEquals(r.buf1().body(), "foo|b");
        assertEquals(r.buf2().body(), "ar|ba");
        assertEquals(r.buf2().trailer().intValue(), 0x7a);
    }
}

File: spec/javascript/test_params_call_short.js
var assert = require('assert');
var testHelper = require('testHelper');

testHelper('ParamsCallShort', 'src/term_strz.bin', function(r) {
  assert.equal(r.buf1.body, "foo|b");
  assert.equal(r.buf2.body, "ar|ba");
  assert.equal(r.buf2.trailer, 0x7a);
});
